src-electron/db/query-endpoint.js

```diff
@@ -58,6 +58,75 @@ ORDER BY E1.ENDPOINT_IDENTIFIER
   return rows.map(dbMapping.map.endpoint)
 }
 
+/**
+ * Returns a promise resolving into all endpoints based on the template
+ * category(eg zigbee/matter).
+ *
+ * @param {*} db
+ * @param {*} sessionId
+ * @param {*} templateCategory
+ * @returns Promise that resolves to endpoints.
+ */
+async function selectAllEndpointsBasedOnTemplateCategory(
+  db,
+  sessionId,
+  templateCategory
+) {
+  let rows = await dbApi.dbAll(
+    db,
+    `
+SELECT
+  E1.ENDPOINT_ID,
+  E1.SESSION_REF,
+  E1.ENDPOINT_TYPE_REF,
+  E1.PROFILE,
+  E1.ENDPOINT_IDENTIFIER,
+  E1.NETWORK_IDENTIFIER,
+  PACKAGE.CATEGORY,
+  E2.ENDPOINT_ID AS PARENT_ENDPOINT_REF,
+  E2.ENDPOINT_IDENTIFIER AS PARENT_ENDPOINT_IDENTIFIER
+FROM
+  ENDPOINT AS E1
+LEFT JOIN
+  ENDPOINT AS E2
```

Comment:
I got confused by now.... Why exactly do we need the parent-child relationship in the endpoints? What does a "parent endpoint" mean?

Reply:
An endpoint which can be a composition of other endpoints. So that relationship is established by parent endpoints